x-pack/examples/embedded_lens_example/public/app.tsx

```diff
@@ -113,19 +92,68 @@ function getLensAttributes(
 export const App = (props: {
   core: CoreStart;
   plugins: StartDependencies;
-  defaultIndexPattern: IndexPattern | null;
+  defaultDataView: DataView | null;
+  lensFormulaHelper: FormulaHelper;
 }) => {
   const [color, setColor] = useState('green');
+  const [dataLayer, setDataLayer] = useState<PersistedIndexPatternLayer>();
   const [isLoading, setIsLoading] = useState(false);
   const [isSaveModalVisible, setIsSaveModalVisible] = useState(false);
-  const LensComponent = props.plugins.lens.EmbeddableComponent;
-  const LensSaveModalComponent = props.plugins.lens.SaveModalComponent;
-
   const [time, setTime] = useState({
     from: 'now-5d',
     to: 'now',
   });
 
+  const LensComponent = props.plugins.lens.EmbeddableComponent;
+  const LensSaveModalComponent = props.plugins.lens.SaveModalComponent;
+
+  useEffect(() => {
```

Comment:
@alexwizp It seems like we don't need the complexity of a hook anymore for the API

Reply:
@flash1293 yes/no. We need a `dataView` for generating `formulaColumns`.  I can definitely do it without hook after checking that 'defaultDataView' is exist.

`if (!props.defaultDataView || !props.defaultDataView.isTimeBased()) {`
   
But from the React perspective we probably should use hooks for that kind of cases. If you insist I'll fix

Reply:
Yes please, let's revert it to the approach that was used before.